### PR TITLE
#174797474 Avail Requests for Approval

### PR DESCRIPTION
**Pull request by jpirumvaa**
**What does this PR do?**
This PR displays the manager's requests that are associated with him

**Description of Task to be completed?**
- create a component for displaying all requests
- create pagination to display a few requests per page
- test the feature 

**How should this be manually tested?**

- clone this repo `git clone https://github.com/atlp-rwanda/script-heroes-frontend`
- checkout the branch git checkout `avail-requests-#174797474 `
- install dependencies` yarn install`
- start the app yarn run dev
- got to` /request/manager` to display all requests 
- check responsiveness on different screen sizes

**Any background context you want to provide?**

> set the token in local storage `x-auth-token: value`

**What are the relevant pivotal tracker stories?**
   [#174797474](https://www.pivotaltracker.com/n/projects/2465784/stories/174797474)

**Screenshots (if appropriate)**
noreq1
noreq

### CODE REVIEW

File: redux/actions/adminRequests.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { GET_REQUESTS, GET_REQUESTS_ERROR} from "./types";
+
+export const adminRequests = () => async(dispatch) => {
+  try {
+    let token= localStorage.getItem('token')
+    const res = await http.get(`/api/request/manager`,{
+      headers:{
+        "x-auth-token": token
+      }
+    });
```
Comment (ericus123) [inline redux/actions/adminRequests.js]: Remove the `x-auth-token `header as it's already included in` axios configuration`. 
Also delete `token` variable,we are using `x-auth-token` header configured in axios .

File: redux/actions/adminRequests.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    dispatch({ 
+      type: GET_REQUESTS, 
+      payload: res.data 
+    });
+  } catch (error) {
+    console.log(error.response);
+    dispatch({
+      type: GET_REQUESTS_ERROR,
+      payload: error.response.data.error,
+    });
```
Comment (ericus123) [inline redux/actions/adminRequests.js]: emprove error handling to handle `authentication` and `authorisation` errors .

File: redux/actions/adminRequests.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import http from "../../src/utils/axios";
+import { GET_REQUESTS, GET_REQUESTS_ERROR} from "./types";
+
+export const adminRequests = () => async(dispatch) => {
+  try {
+    const res = await http.get(`/api/request/manager`);
+    console.log("The response is:",res.data)
```
Comment (elvisiraguha) [inline redux/actions/adminRequests.js]: You might need to remove this _console.log_ if it was used for debugging purposes

**Review by elvisiraguha — APPROVED** [review]
LGTM

**Review by dushimeemma — APPROVED** [review]
Good job. but you need to import Login component from views instead of Views

**Review by ericus123 — APPROVED** [review]
Good job.
Delete the comments in tests if no longer needed

**Review by Adadie — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]

**Pull request merged**